File: CGStudio/FluidStudioTest/SolverSceneTest.cs
using System;
using System.Collections.Generic;
using FluidStudio.Physics;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using PG.Core.Math;
using PG.Scene;

namespace FluidStudioTest
{
    [TestClass]
    public class SolverSceneTest
    {
        private FluidScene CreateFluidTestScene(SceneList world, string name)
        {
            var particles = new List<Vector3d>();
            particles.Add(new Vector3d(0, 0, 0));
            var id = world.AddParticleSystemScene(particles, "", new PG.Core.UI.ParticleAppearance(), 1);
            return new FluidScene(world, id, 1.0f, 1.0f, name, false);
        }

        private SolverScene CreateSolverTestScene(SceneList world, string name)
        {
            var fluids = new List<FluidScene>();
            fluids.Add(CreateFluidTestScene(world, "Fluid"));
            var boundaries = new List<CSGBoundaryScene>();

            var solver = new SolverScene();
            solver.Create(world, fluids, boundaries, 0.03f, name);
            return solver;
        }

        [TestMethod]
        public void TestCreate()
        {
            var world = new SceneList();
            var solver = CreateSolverTestScene(world, "");
            Assert.AreEqual(3, solver.Id);
        }

        [TestMethod]
        public void TestSimulate()
        {
            var world = new SceneList();
            var solver = CreateSolverTestScene(world, "");
            solver.Simulate(world);
        }
    }
}

File: Blender/CrystalPythonTest/physics/solver_scene_test.py
import unittest
from scene.particle_system_scene import *
from physics.fluid_scene import *
from physics.solver_scene import *
from physics.csg_boundary_scene import CSGBoundaryScene
from CrystalPLI import Vector3df

class SolverSceneTest(unittest.TestCase):
    def test(self):
        world = World()
        scene = Scene(world)
        solver = SolverScene(scene)
        solver.create()
        self.assertEqual(1, solver.id)

        ps = ParticleSystemScene(scene)
        ps.create_empty("")

        fluids = []
        fluid = FluidScene(scene)
        fluid.create()
        fluid.source_particle_system_id = ps.id
        fluid.send()
        fluids.append(fluid)

        boundaries = []
        boundary = CSGBoundaryScene(scene)
        boundary.create()
        boundary.send()
        boundaries.append(boundary)

        solver.fluids = fluids
        solver.boundaries = boundaries
        solver.external_force = Vector3df(0.0, 0.0, -9.8)
        solver.send()
        solver.simulate()

        solver.export_pcd("export_macro.pcd", False)
        solver.export_pcd("export_micro.pcd", True)
